### PR TITLE
feat(oracle) Add oracle linux oval

### PR DESCRIPTION
**Pull request by masahiro331**
Support Oracle Linux Security Advisory.
related: https://github.com/aquasecurity/vuln-list-update/pull/18

# Note
## Source Data
```
{
  "Title": "\nELSA-2007-0123:  Moderate: cups security update  (MODERATE)\n",
  "Description": "\n [1.1.22-0.rc1.9.18]\n - REVERTED these changes:\n   - Applied patch from STR #1301 (bug #195354).\n   - Patch pdftops to understand 'includeifexists', and use that in the\n     pdftops.conf file (bug #188583).\n   - Clear the printer's state_message and state_reasons after successful\n     job completion (bug #187457).\n   - Include dest-cache-v2 patch (bug #175847).\n   - Back-ported CUPS 1.2.x change to fix out of order IPP jobs (bug #171142).\n   - Back-ported large file support (bug #211915).\n   - Back-ported HTTP timing fix for STR #1020 (bug #194025).\n \n [1.1.22-0.rc1.9.16]\n - Restored use_dbus setting.\n \n [1.1.22-0.rc1.9.15]\n - Added timeouts to SSL negotiation (bug #232241).\n \n [1.1.22-0.rc1.9.14]\n - Back-ported HTTP timing fix for STR #1020 (bug #194025).\n \n [1.1.22-0.rc1.9.13]\n - Back-ported large file support (bug #211915).\n \n [1.1.22-0.rc1.9.12]\n - Back-ported CUPS 1.2.x change to fix out of order IPP jobs (bug #171142).\n - Include dest-cache-v2 patch (bug #175847).\n - Resolves: rhbz #171142 \n",
  "Platform": [
    "Oracle Linux 5"
  ],
  "References": [
    {
      "Source": "elsa",
      "URI": "http://linux.oracle.com/errata/ELSA-2007-0123.html",
      "ID": "ELSA-2007-0123"
    },
    {
      "Source": "CVE",
      "URI": "http://linux.oracle.com/cve/CVE-2007-0720.html",
      "ID": "CVE-2007-0720"
    }
  ],
・・・ // Cut Criteria info
  "Severity": "MODERATE",
  "CVEs": [
    {
      "Impact": "",
      "Href": "http://linux.oracle.com/cve/CVE-2007-0720.html",
      "ID": "CVE-2007-0720"
    }
  ]
}
```

## FullDB  (boltbrower view)
```
Path: vulnerability → ELSA-2007-0123
Key: ELSA-2007-0123
Value:
{
  "Description": "\n [1.1.22-0.rc1.9.18]\n - REVERTED these changes:\n   - Applied patch from STR #1301 (bug #195354).\n 
  "References": [
    "http://linux.oracle.com/cve/CVE-2007-0720.html",
    "http://linux.oracle.com/errata/ELSA-2007-0123.html"
  ],
  "Severity": "MEDIUM",
  "Title": "ELSA-2007-0123:  Moderate: cups security update  (MODERATE)\n"
}
```

## LightDB  (boltbrower view)
```
Path: severity → ELSA-2007-0123
Key: ELSA-2007-0123
Value: MEDIUM
```

### CODE REVIEW

File: pkg/vulnsrc/oracle-oval/oracle-oval.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+func NewVulnSrc() VulnSrc {
+	return VulnSrc{
+		dbc: db.Config{},
+	}
+}
+
+func (vs VulnSrc) Update(dir string) error {
+	rootDir := filepath.Join(dir, "vuln-list", oracleDir)
+
+	var cves []OracleOVAL
```
[the remark this answers is not in the document]
Comment (masahiro331) [reply pkg/vulnsrc/oracle-oval/oracle-oval.go]: The "advisory" variable uses another line.
https://github.com/aquasecurity/trivy-db/pull/8/files#diff-b1f68de92b47e1d58e0671d690dccca0R81

So change cves-> ovals, cve-> oval.

File: pkg/vulnsrc/oracle-oval/oracle-oval_test.go [hunk 2]
```diff
@@ -0,0 +1,873 @@
...
+						Criterias: []Criteria{
+							{
+								Operator: "OR",
+								Criterias: []Criteria{
+									{
+										Operator:  "AND",
+										Criterias: nil,
+										Criterions: []Criterion{
+											{
+												Comment: "bind-devel is earlier than 30:9.3.3-8.el5",
```
Comment (knqyf263) [inline pkg/vulnsrc/oracle-oval/oracle-oval_test.go]: Would you add other packages? e.g. bind-libs, bind-chroot.

File: pkg/vulnsrc/oracle-oval/oracle-oval_test.go [hunk 3]
```diff
@@ -0,0 +1,873 @@
...
+											},
+										},
+									},
+								},
+								Criterions: nil,
+							},
+						},
+						Criterions: []Criterion{
+							{
+								Comment: "Oracle Linux 5 is installed",
```
Comment (knqyf263) [inline pkg/vulnsrc/oracle-oval/oracle-oval_test.go]: Would you add a case including two major versions? e.g. Oracle Linux 5 and Oracle Linux 6.

File: pkg/vulnsrc/oracle-oval/oracle-oval.go [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+		}
+		if ss[1] == "0" {
+			continue
+		}
+
+		pkgs = append(pkgs, AffectedPackage{
+			OSVer: osVer,
+			Package: Package{
+				Name:         ss[0],
+				FixedVersion: version.NewVersion(strings.Split(ss[1], " ")[0]).String(),
```
Comment (knqyf263) [inline pkg/vulnsrc/oracle-oval/oracle-oval.go]: Why does it need `strings.Split`?

File: pkg/vulnsrc/oracle-oval/oracle-oval.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+			},
+		})
+	}
+
+	if len(cri.Criterias) == 0 {
```
Comment (knqyf263) [inline pkg/vulnsrc/oracle-oval/oracle-oval.go]: Are these lines necessary?

File: pkg/vulnsrc/oracle-oval/oracle-oval.go [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+	return nil
+
+}
+
+func (vs VulnSrc) commit(tx *bolt.Tx, ovals []OracleOVAL) error {
+	for _, oval := range ovals {
+		elsaID := strings.Split(oval.Title, ":")[0]
+
+		nonCvesFlag := false
```
Comment (knqyf263) [inline pkg/vulnsrc/oracle-oval/oracle-oval.go]:
```suggestion
		var vulnIDs []string
		for _, cve := range oval.Cves {
			vulnIDs = append(vulnIDs, cve.ID)
		}
		if len(vulnIDs) == 0 {
			vulnIDs = append(vulnIDs, elsaID)
		}
```

File: pkg/utils/utils_test.go [hunk 7]
```diff
@@ -82,3 +83,54 @@ func TestFileWalk(t *testing.T) {
...
+			expectData: []string{
+				"test string 1",
+				"test string 2",
+				"test string 3",
+			},
+		},
+	}
+	for i, testCase := range testCases {
+		actualData := Uniq(testCase.inputData)
+		if !reflect.DeepEqual(actualData, testCase.expectData) {
```
Comment (knqyf263) [inline pkg/utils/utils_test.go]: Would you use testify/assert?

**Pull request merged**